tests/functional/a/arguments_differ.py

```diff
@@ -326,3 +326,34 @@ class Foo2(AbstractFoo):
 
     def kwonly_6(self, first, *args, **kwargs):  # valid override
         "One positional with the rest variadics to pass through parent params"
+
+
+# Adding arguments with default values to a child class is valid
+# See:
+# https://github.com/PyCQA/pylint/issues/1556
+# https://github.com/PyCQA/pylint/issues/5338
+
+
+class BaseClass:
+    def method(self, arg: str):
+        print(self, arg)
+
+
+class DerivedClassWithAnnotation(BaseClass):
+    def method(self, arg: str, param1: int = 42, *, param2: int = 42):
+        print(arg, param1, param2)
+
+
+class DerivedClassWithoutAnnotation(BaseClass):
+    def method(self, arg, param1=42, *, param2=42):
+        print(arg, param1, param2)
+
+
+class AClass:
+    def method(self, *, arg1):  # [no-self-use]
+        ...

```

Comment:
Is it the usual practice to mix in unrelated messages instead of using correct style or suppressing? A `print(self)` is not _that_ much uglier than `...`

Reply:
I think generally it's a balance between functional tests that are unitary and having a lot of code construct and example without too much duplication. For no-self-use I don't think we need more tests for it. Also we're probably going to remove it from the default configuration in #3512 so I think using print(self) will still work then but we'd get a useless-suppression if we disable or a "expected a message" from the test with the current code.

Reply:
Yeah, I was just lazy here. Added a print.